### DISCUSSION TITLE
Does resip is support the RFC 8760?

### DISCUSSION THREAD

**Discussion by portsip** [resiprocate/resiprocate#318] [Q&A] [answered]
Hi team,

I assume the resip is not supporting the RFC 8760 yet, any plan for this?

Thanks

**Comment by sgodin** [comment, marked as answer]
Just skimmed the RFC, it is NOT currently supported by resip.  I'm not aware of anyone's plan to add this feature.

Scott

**Comment by dilkie** [comment]
yeah, we're interested in this as well, some customers are asking for it.

**Comment by dilkie** [comment]
BTW, I did add support for this (SHA256/SHA512-256 with) to a cloned copy of resip so we could support it but it uses openssl, which resip does not depend on as it rolls it's own cipher code..

**Comment by sgodin** [comment]
Hmm interesting.  Resip does build with openssl, the resip TLS transport and other stack features depend on it.  If OpenSSL support is not asked for, then many features of resip that need it, get disabled.    We could put all the code for this behind USE_SSL preprocessor defines, so it only works if OpenSSL support is enabled.  Just search for USE_SSL in the code base for examples.  Would you be willing to put a PR together for this?

**Comment by dilkie** [comment]
really? It didn't look so in the code I was looking at. It had it's own MD5 implementation.  Note, all the work i did was in resip/stack. I created a generic "DigestStream" (borrowed from your rutil/MD5Stream) that takes a string arg ("MD5, SHA1", "SHA256", "SHA512-256") and adjusted some code to use it. Your md5 code is resident in rutil/vmd5.cxx/hxx.

I can share my diff (off resiprocate-1.12) if you want.

**Comment by sgodin** [comment]
You are correct about some code, there are some non-openssl implemented algorithms.  We could continue to use those if USE_SSL is disabled, and use your new code if USE_SSL is enabled.  So we would only end up advertising support for SHA authentication if USE_SSL is enabled.

**Comment by sgodin** [comment]
@dilkie I'm looking at getting this officially supported in resip, can you please share your implementation code with me?
Here is my email if you prefer:  sgodin@sipspectrum.com
Thank you!

**Comment by sgodin** [comment]
Official SHA256 and SHA512/256 support has landed in resip!  Please see this wiki page for details:
https://github.com/resiprocate/resiprocate/wiki/Digest-Authentication-SHA256-and-SHA512%E2%80%90256-Support
